procedures/CodeBrowser.ipf

```diff
@@ -211,6 +217,38 @@ Function/S interpretParameters(funcInfo)
 	return str
 End
 
+// check if Function has as a WM*Struct as first parameter
+//
+// handles the following parameters:
+// * WMWinHookStruct
+// * WMBackgroundStruct
+// * WM.*Struct
+//
+// @param[in]  funcInfo        output of FunctionInfo for the function in question
+// @param[out] WMDotStarStruct matched type of the structure as string.
+//                             Not changed if 0 was returned.
+//
+// @returns 1 if function has such a parameter, 0 otherwise
+Function FunctionWithWMstruct(funcInfo, WMDotStarStruct)
+	string funcInfo
+	string &WMDotStarStruct
+
+	string declaration, re, typeName
+	string handledWMStructs = "WinHook|Background|[^\s,]+" // WM + handledWMStructs + Struct
+
+	if(NumberByKey("PARAM_0_TYPE", funcInfo) & 0x200 | 0x1000) // struct | pass-by-reference
+		declaration = getFunctionLine(1, funcInfo)
+		re = "(?i)\s*struct\s+(WM(?:" + handledWMStructs + ")Struct)\s+.*"
+		SplitString/E=(re) declaration, typeName
```

Comment:
How about making it viable for all structures? You already proposed a regexp which I used

```
$ git diff .
diff --git a/procedures/CodeBrowser.ipf b/procedures/CodeBrowser.ipf
index 98d78a8..ecf9105 100644
--- a/procedures/CodeBrowser.ipf
+++ b/procedures/CodeBrowser.ipf
@@ -234,11 +234,10 @@ Function FunctionWithWMstruct(funcInfo, WMDotStarStruct)
        string &WMDotStarStruct

        string declaration, re, typeName
-       string handledWMStructs = "WinHook|Background|[^\s,]+" // WM + handledWMStructs + Struct

-       if(NumberByKey("PARAM_0_TYPE", funcInfo) & 0x200 | 0x1000) // struct | pass-by-reference
+       if(NumberByKey("PARAM_0_TYPE", funcInfo) & (0x200 | 0x1000)) // struct | pass-by-reference
                declaration = getFunctionLine(1, funcInfo)
-               re = "(?i)\s*struct\s+(WM(?:" + handledWMStructs + ")Struct)\s+.*"
+               re = "(?i)^\s*struct\s+(\w+)\s+"
                SplitString/E=(re) declaration, typeName
                if(V_flag == 1)
                        WMDotStarStruct = typeName

```

and that works for

```
Structure MyStruct
EndStructure

Function ConsumeStruct(s)
        STRUCT MyStruct &s

        return 0
End
```

already.

In my patch I also added some parantheses, the precedence rules for `&` and `|` are a bit difficult to know by heart.

Reply:
it works without matching the last `.*` so I kept it like you proposed.